src/Calculator/Views/CalculatorProgrammerBitFlipPanel.xaml.cpp

```diff
@@ -185,35 +174,51 @@ void CalculatorProgrammerBitFlipPanel::UpdateCheckedStates()
         return;
     }
 
-    static const wchar_t ch0 = LocalizationSettings::GetInstance().GetDigitSymbolFromEnUsDigit(L'0');
-
-    // Filter any unwanted characters from the displayed string.
-    static constexpr array<wchar_t, 4> unwantedChars = { L' ', Utils::LRE, Utils::PDF, Utils::LRO };
-
-    wstringstream stream;
-    wstring displayValue = Model->BinaryDisplayValue->Data();
-    for (const wchar_t& c : displayValue)
+    m_updatingCheckedStates = true;
+    unsigned int bitIndex = 0;
+    auto it = m_flipButtons.begin();
+    for (bool val : Model->BinaryDigits)
     {
-        if (find(begin(unwantedChars), end(unwantedChars), c) == unwantedChars.end())
+        // Highest bit (64) is at index 0 in bit string.
+        // To get bit 0, grab from opposite end of string.
```

Comment:
I think this comment makes more sense in StandardCalculatorViewModel.cpp where you actually reverse the string.  I was a little confused when reading this as to why you were using reverse iterators until I saw this comment here.